### PR TITLE
[glyphs2fontir] Fallback to setting GlobalMetrics from font

### PR DESCRIPTION
**Pull request by khaledhosny**
If the corresponding custom parameter is missing from the master, try it from the parent font.

- Fixes https://github.com/googlefonts/fontc/issues/949

### CODE REVIEW

File: fontir/src/ir.rs [hunk 1]
```diff
@@ -664,6 +664,20 @@ impl GlobalMetrics {
         }
     }
 
+    pub fn set_if_some_or_fallback(
+        self: &mut GlobalMetrics,
```
Comment (cmyr) [inline fontir/src/ir.rs:668]: I'm... not sure why this is like this above, but this signature is just a weird way of writing `&mut self` as the first param, and we should stick to that.

(unless I'm missing something)

File: glyphs2fontir/src/source.rs [hunk 2]
```diff
@@ -545,105 +585,125 @@ impl Work<Context, WorkId, Error> for GlobalMetricWork {
 
             metrics.set_if_some(GlobalMetric::CapHeight, pos.clone(), master.cap_height());
             metrics.set_if_some(GlobalMetric::XHeight, pos.clone(), master.x_height());
-            metrics.set_if_some(
+            metrics.set_if_some_or_fallback(
                 GlobalMetric::Os2TypoAscender,
                 pos.clone(),
                 master.typo_ascender.map(|v| v as f64),
+                font.typo_ascender.map(|v| v as f64),
```
Comment (cmyr) [inline glyphs2fontir/src/source.rs:592]: okay so the other way to write all of these would be to stick with `set_if_some`, but then do,

```rust
metrics.set_if_some(
    MetricName,
    pos.clone(),
    master.typo_ascender.or(font.typo_ascender).map(|v| v as f64),
)
```

which might be... slightly better, on balance? at least it saves us adding the additional method. This whole big function is kind of calling out for a macro, but that's another story; let's just get it working first.

Comment (khaledhosny) [reply glyphs2fontir/src/source.rs:592]: Sounds good. I’ll try to do this in the morning.

File: fontir/src/ir.rs (hunk 1 above)
Comment (khaledhosny) [reply fontir/src/ir.rs:668]: I just copied the above function and added a new parameter without giving it much thought.

**Review by cmyr — APPROVED** [review]
thanks!

**Pull request merged**